spack test run tasmanian (compiler version needs to be specified)

```bash
$ spack test run tasmanian
==> Spack test z55haf5f7dbvpcrnzl4ski2lgl243s4g
==> Testing package tasmanian-7.7-o54geas
==> Error: ProcessError: Command exited with status 1:
    'cmake' '/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/tasmanian-7.7-o54geasisuouzsyuaeh3fyxbrqzjknsl/share/Tasmanian/testing'

1 error found in test log:
  >> 3    CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
     4      CMake 3.21 or higher is required.  You are running version 3.16.3
     5    
     6    
     7    -- Configuring incomplete, errors occurred!

See test log for details:
  /home/vanessa/.spack/test/z55haf5f7dbvpcrnzl4ski2lgl243s4g/tasmanian-7.7-o54geas-test-out.txt

==> Error: 1 test(s) in the suite failed.
```

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -173,12 +173,10 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
 
     @run_after('install')
     def setup_smoke_test(self):
-        if not self.spec['cmake'].satisfies('@3.22:'):
-            tty.msg('Error tasmanian test: CMake 3.21 or higher is required')
+        if not self.spec['cmake'].satisfies('@3.10:'):
+            tty.msg('Error tasmanian test: CMake 3.10 or higher is required')
             return
 
-        install_tree(self.prefix.share.Tasmanian.examples,
-                     join_path(self.install_test_root, 'examples'))
         install_tree(self.prefix.share.Tasmanian.testing,
                      join_path(self.install_test_root, 'testing'))
         self.cmake_bin(set=True)

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -4,7 +4,10 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from llnl.util import tty
+
 from spack import *
+
+import os
 
 
 class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
@@ -160,31 +163,32 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
     def cmake_bin(self, set=True):
         """(Hack) Set/get cmake dependency path."""
         filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
-        if set:
-            with open(filepath, 'w') as out_file:
-                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
-                out_file.write('{0}\n'.format(cmake_bin))
+        if os.path.isfile(filepath):
+            if set:
+                mkdirp(self.install_test_root)
+                with open(filepath, 'w') as out_file:
+                    cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                    out_file.write('{0}\n'.format(cmake_bin))
+            else:
+                with open(filepath, 'r') as in_file:
+                    return in_file.read().strip()
         else:
-            with open(filepath, 'r') as in_file:
-                return in_file.read().strip()
+            tty.msg('Skipping tasmanian test: cmake_bin_path.txt not found')
 
     @run_after('install')
     def setup_smoke_test(self):
-        self.cmake_bin(set=True)
+        if self.spec['cmake'].satisfies('@3.22:'):
+            self.cmake_bin(set=True)
 
     def test(self):
         cmake_bin = self.cmake_bin(set=False)
 
-        if self.spec['cmake'].satisfies('@3.22:'):
-            self.run_test(cmake_bin,
-                          options=self.cmake_args(),
-                          purpose='Build with same CMake version as install')
+        self.run_test(cmake_bin,
+                      purpose='Build with same CMake version as install')
 
-            # using the tests installed in <prefix>/share/Tasmanian/testing
-            cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
-            with working_dir(self.test_suite.current_test_cache_dir, create=True):
-                cmake(cmake_dir)
-                make()
-                make('test')
-        else:
-            tty.msg('Skipping tasmanian test: CMake 3.21 or higher is required')
+        # using the tests installed in <prefix>/share/Tasmanian/testing
+        cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
+        with working_dir(self.test_suite.current_test_cache_dir, create=True):
+            cmake(cmake_dir)
+            make()
+            make('test')

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -194,17 +194,17 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
         cmake_dir = self.test_suite.current_test_cache_dir.testing
 
         if not self.run_test(cmake_bin,
-                      options=[cmake_dir],
-                      purpose='Generate the Makefile'):
+                             options=[cmake_dir],
+                             purpose='Generate the Makefile'):
             tty.msg('Skipping tasmanian test: failed to generate Makefile')
             return
 
         if not self.run_test('make',
-                      purpose='Build test software'):
+                             purpose='Build test software'):
             tty.msg('Skipping tasmanian test: failed to build test')
             return
 
         if not self.run_test('make',
-                      options=['test'],
-                      purpose='Run test'):
+                             options=['test'],
+                             purpose='Run test'):
             tty.msg('Failed tasmanian test: failed to run test')

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -3,11 +3,11 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
+
 from llnl.util import tty
 
 from spack import *
-
-import os
 
 
 class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
@@ -163,32 +163,39 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
     def cmake_bin(self, set=True):
         """(Hack) Set/get cmake dependency path."""
         filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
-        if os.path.isfile(filepath):
-            if set:
-                mkdirp(self.install_test_root)
-                with open(filepath, 'w') as out_file:
-                    cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
-                    out_file.write('{0}\n'.format(cmake_bin))
-            else:
-                with open(filepath, 'r') as in_file:
-                    return in_file.read().strip()
-        else:
-            tty.msg('Skipping tasmanian test: cmake_bin_path.txt not found')
+        if set:
+            with open(filepath, 'w') as out_file:
+                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                out_file.write('{0}\n'.format(cmake_bin))
+        elif os.path.isfile(filepath):
+            with open(filepath, 'r') as in_file:
+                return in_file.read().strip()
 
     @run_after('install')
     def setup_smoke_test(self):
-        if self.spec['cmake'].satisfies('@3.22:'):
-            self.cmake_bin(set=True)
+        if not self.spec['cmake'].satisfies('@3.22:'):
+            return
+
+        install_tree(self.prefix.share.Tasmanian.examples,
+                     join_path(self.install_test_root, 'examples'))
+        install_tree(self.prefix.share.Tasmanian.testing,
+                     join_path(self.install_test_root, 'testing'))
+        self.cmake_bin(set=True)
 
     def test(self):
         cmake_bin = self.cmake_bin(set=False)
 
-        self.run_test(cmake_bin,
-                      purpose='Build with same CMake version as install')
+        if not cmake_bin:
+            tty.msg('Skipping tasmanian test: cmake_bin_path.txt not found')
+            return
 
         # using the tests installed in <prefix>/share/Tasmanian/testing
-        cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
-        with working_dir(self.test_suite.current_test_cache_dir, create=True):
-            cmake(cmake_dir)
-            make()
-            make('test')
+        cmake_dir = self.test_suite.current_test_cache_dir.testing
+        self.run_test(cmake_bin,
+                      options=[cmake_dir],
+                      purpose='Generate the Makefile')
+        self.run_test('make',
+                      purpose='Build test software')
+        self.run_test('make',
+                      options=['test'],
+                      purpose='Run test')

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -204,8 +204,7 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
             tty.msg('Skipping tasmanian test: failed to build test')
             return
 
-       if not self.run_test('make',
+        if not self.run_test('make',
                       options=['test'],
                       purpose='Run test'):
-            tty.msg('Failed tasmanian test')
-            return
+            tty.msg('Failed tasmanian test: failed to run test')

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -174,6 +174,7 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
     @run_after('install')
     def setup_smoke_test(self):
         if not self.spec['cmake'].satisfies('@3.22:'):
+            tty.msg('Error tasmanian test: CMake 3.21 or higher is required')
             return
 
         install_tree(self.prefix.share.Tasmanian.examples,
@@ -189,13 +190,22 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
             tty.msg('Skipping tasmanian test: cmake_bin_path.txt not found')
             return
 
-        # using the tests installed in <prefix>/share/Tasmanian/testing
+        # using the tests copied from <prefix>/share/Tasmanian/testing
         cmake_dir = self.test_suite.current_test_cache_dir.testing
-        self.run_test(cmake_bin,
+
+        if not self.run_test(cmake_bin,
                       options=[cmake_dir],
-                      purpose='Generate the Makefile')
-        self.run_test('make',
-                      purpose='Build test software')
-        self.run_test('make',
+                      purpose='Generate the Makefile'):
+            tty.msg('Skipping tasmanian test: failed to generate Makefile')
+            return
+
+        if not self.run_test('make',
+                      purpose='Build test software'):
+            tty.msg('Skipping tasmanian test: failed to build test')
+            return
+
+       if not self.run_test('make',
                       options=['test'],
-                      purpose='Run test')
+                      purpose='Run test'):
+            tty.msg('Failed tasmanian test')
+            return

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -3,6 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from llnl.util import tty
 from spack import *
 
 
@@ -153,10 +154,37 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
 
         return args
 
+    # TODO: Replace this method and its 'get' use for cmake path with
+    #   join_path(self.spec['cmake'].prefix.bin, 'cmake') once stand-alone
+    #   tests can access build dependencies through self.spec['cmake'].
+    def cmake_bin(self, set=True):
+        """(Hack) Set/get cmake dependency path."""
+        filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
+        if set:
+            with open(filepath, 'w') as out_file:
+                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                out_file.write('{0}\n'.format(cmake_bin))
+        else:
+            with open(filepath, 'r') as in_file:
+                return in_file.read().strip()
+
+    @run_after('install')
+    def setup_smoke_test(self):
+        self.cmake_bin(set=True)
+
     def test(self):
-        # using the tests installed in <prefix>/share/Tasmanian/testing
-        cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
-        with working_dir(self.test_suite.current_test_cache_dir, create=True):
-            cmake(cmake_dir)
-            make()
-            make('test')
+        cmake_bin = self.cmake_bin(set=False)
+
+        if self.spec['cmake'].satisfies('@3.22:'):
+            self.run_test(cmake_bin,
+                          options=self.cmake_args(),
+                          purpose='Build with same CMake version as install')
+
+            # using the tests installed in <prefix>/share/Tasmanian/testing
+            cmake_dir = join_path(self.prefix, 'share', 'Tasmanian', 'testing')
+            with working_dir(self.test_suite.current_test_cache_dir, create=True):
+                cmake(cmake_dir)
+                make()
+                make('test')
+        else:
+            tty.msg('Skipping tasmanian test: CMake 3.21 or higher is required')

